ui/src/dashboards/actions/index.ts

```diff
@@ -262,6 +298,46 @@ export const createDashboard = () => async (
   }
 }
 
+export const cloneUtilFunc = async (dash: Dashboard, id: string) => {
```

Comment:
this utility function is based on the client-api:

https://github.com/influxdata/influxdb-client-js/blob/master/src/wrappers/dashboards.ts#L247-L248

My first attempt at this refactor tried to flatten these into a synchronous progression, but the functionality was buggy.